backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java

```diff
@@ -37,6 +37,80 @@
 
 public class ElasticSearchConfiguration {
 
+    public enum HostScheme {
+        HTTP("http"),
+        HTTPS("https");
+
+        public static Optional<HostScheme> maybe(String schemeAsString) {
```

Comment:
I think that it breaks our principles of trying to get rid of `null` in the codebase. The caller can wrap it in an optional if it wants to express emptiness